src/proto/single/mod.rs

```diff
@@ -83,10 +128,41 @@ pub struct Job {
     /// across a complex distributed system, etc.
     #[serde(skip_serializing_if = "HashMap::is_empty")]
     #[serde(default = "HashMap::default")]
+    #[builder(default = "HashMap::default()")]
     pub custom: HashMap<String, serde_json::Value>,
 }
 
-#[derive(Serialize, Deserialize, Debug)]
+impl JobBuilder {
+    /// Setter for the arguments provided for this job.
+    pub fn args<A>(&mut self, args: Vec<A>) -> &mut Self
+    where
+        A: Into<serde_json::Value>,
+    {
+        self.args = Some(args.into_iter().map(|s| s.into()).collect());
+        self
+    }
+
+    fn validate(&self) -> Result<(), String> {
+        if let Some(ref priority) = self.priority {
+            if *priority > Some(JOB_PRIORITY_MAX) {
+                return Err("`priority` must be in the range from 0 to 9 inclusive".to_string());
+            }
+        }
```

Comment:
If this is the only kind of misconfiguration there can be for a job, I'm inclined to not have it return an error (and thus get rid of the new error type and the `Result` in `build`'s return type. Especially because `priority` is already a `pub` field, so users can always set it to the wrong value elsewhere.

Reply:
Got rid of this.